File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i=0
        j=len(height)-1
        l = []
        while(i<=j):
            a = height[i]
            b = height[j]
            hei = min(a,b)
            wid = j-i
            area = hei*wid
            l.append(area)
            if(a<=b):
                i+=1
            else:
                j-=1
        return max(l)
            

            

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int j = height.size() - 1;
        vector<int> l;

        while (i <= j) {
            int a = height[i];
            int b = height[j];
            int hei = std::min(a, b);
            int wid = j - i;
            int area = hei * wid;
            l.push_back(area);
            
            if (a <= b) {
                i++;
            } else {
                j--;
            }
        }

        return *max_element(l.begin(), l.end());
    }
};